workers/data_refinery_workers/processors/salmon.py

```diff
@@ -87,12 +84,28 @@ def _prepare_files(job_context: Dict) -> Dict:
         job_context["success"] = False
         return job_context
 
+    if len(original_files) == 2:
+        job_context["input_file_path_2"] = original_files[1].absolute_file_path
+    if not os.path.exists(job_context["input_file_path_2"]):
```

Comment:
This if statement is assuming that the job_context has had the `input_file_path_2` key set, but it is valid for that to not be populated. This if should just start with a `job_context.get("input_file_path_2", False) and` like you have below.

Reply:
Or it could just be nested within the `if len(original_files) == 2` conditional block.